db/go/20200929144301_add_vscode_workspace_template.go

```
@@ -0,0 +1,124 @@
+#       - SLACK_USERNAME=onepanel SLACK_TITLE="Your workspace is ready" SLACK_ICON=https://www.gravatar.com/avatar/5c4478592fe00878f62f0027be59c1bd SLACK_MESSAGE="Your workspace is now running" ./slack-notify
+#       command:
+#       - sh
+#       - -c
+`
+
+const vscodeWorkspaceTemplateName = "VisualStudioCode"
+
+func initialize20200929144301() {
+	if _, ok := initializedMigrations[20200929144301]; !ok {
+		goose.AddMigration(Up20200929144301, Down20200929144301)
+		initializedMigrations[20200929144301] = true
+	}
+}
+
+// Up20200929144301 adds Visual Studio Code as a workspace template.
+func Up20200929144301(tx *sql.Tx) error {
+	client, err := getClient()
+	if err != nil {
+		return err
+	}
+	defer client.DB.Close()
+
+	migrationsRan, err := getRanSQLMigrations(client)
+	if err != nil {
+		return err
+	}
+
+	if _, ok := migrationsRan[20200929144301]; ok {
+		return nil
+	}
+
+	namespaces, err := client.ListOnepanelEnabledNamespaces()
+	if err != nil {
+		return err
+	}
+
+	workspaceTemplate := &v1.WorkspaceTemplate{
+		Name:     vscodeWorkspaceTemplateName,
+		Manifest: vscodeWorkspaceTemplate,
+	}
+
+	// Adding description
+	workspaceTemplate.Description = "Interactive development environment for code, notebooks, and everything in between."
+
+	for _, namespace := range namespaces {
+		if _, err := client.CreateWorkspaceTemplate(namespace.Name, workspaceTemplate); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Down20200929144301 removes Visual Studio Code from workspace templates.
+func Down20200929144301(tx *sql.Tx) error {
+	client, err := getClient()
+	if err != nil {
+		return err
+	}
+
+	namespaces, err := client.ListOnepanelEnabledNamespaces()
+	if err != nil {
+		return err
+	}
+	uid, err := uid2.GenerateUID(vscodeWorkspaceTemplateName, 30)
+	if err != nil {
+		return err
+	}
+	for _, namespace := range namespaces {
+		if _, err := client.ArchiveWorkspaceTemplate(namespace.Name, uid); err != nil {
+			log.Fatalf("error %v", err.Error())
```

@aleksandrmelnikov Let's return the err, goose expects that.

---

Agreed, fixed.